framework/tlvf/src/src/CmduTlvParser.cpp

```diff
@@ -108,7 +108,7 @@ std::shared_ptr<BaseClass> CmduTlvParser::parseNextTlv()
     case (17): {
         auto tlv = cmdu_.addClass<ieee1905_1::tlvWsc>();
         if (tlv)
-            WSC::AttrList::parse(*tlv);
+            tlv->addInnerClassList(WSC::AttrList::parse(*tlv));
         return tlv;
     }
```

Comment:
CmduTlvParser shouldn't handle special cases such as this one, the attribute list parsing should be done as part of `tlvWsc::addClass()`, this requires a TLVF change, as tlvWsc is an autogenerated class.
It's still worth mentioning in my opinion, and I will create an issue for that.

Reply:
I generally agree, TLVF should add meta option to describe which inner class list should be part of each TLV and then we can autogenerate the parser to call the `addInnerClassList` but Its an overkill IMO at this stage.